### PR TITLE
fix: create optional dependency group for umap-learn and fast-hdbscan

### PR DESCRIPTION
**Pull request by RogerHYang**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -99,6 +96,11 @@ dev = [
   "uvloop; platform_system != 'Windows'",
   "grpc-interceptor[testing]",
 ]
+umap = [
```
Comment (mikeldking) [inline pyproject.toml:99]: Maybe embeddings?

Comment (mikeldking) [reply pyproject.toml:99]: Also need to add this in the docker build for now

File: pyproject.toml [hunk 2]
```diff
@@ -25,9 +25,6 @@ dependencies = [
   "numpy!=2.0.0",  # https://github.com/lmcinnes/umap/issues/1138
   "pandas>=1.0",
   "jinja2",
-  "umap-learn",
```
Comment (mikeldking) [inline pyproject.toml:28]: Lots of notebooks need updating. Maybe we add the extra then remove from core deps?

File: pyproject.toml (hunk 1 above)
Comment (RogerHYang) [reply pyproject.toml]: done

File: pyproject.toml (hunk 2 above)
Comment (RogerHYang) [reply pyproject.toml]: reverted

**Review by mikeldking — APPROVED** [review]

**Pull request merged**